### ISSUE TITLE
Error when templates were refreshed

### ISSUE THREAD

**Issue by crutkas** [Microsoft/WindowsTemplateStudio#614]
Nightly build, was going through looking for bugs and got this notification that templates were being updated then it threw an error.

![image](https://user-images.githubusercontent.com/1462282/27885486-0ed37eea-618d-11e7-9274-8782849d2100.png)

````
Windows Template Studio 2017-07-05 14:14:58.766
Version: 0.14.17186.65008
>
[02:14:58.661 - Information]::Configuration file loaded: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\PROFESSIONAL\COMMON7\IDE\EXTENSIONS\P22WACVZ.F2A\WindowsTemplateStudio.config.json
[02:14:59.940 - Verbose]::Templates content downloaded to C:\Users\crutkas\AppData\Local\Temp\3uyvhb0v.gci\Templates.mstx from https://wtsrepository.blob.core.windows.net/dev/Latest/Templates.mstx.
[02:14:59.969 - Verbose]::Package certificate OID.0.9.2342.19200300.100.1.1=10018646, CN=Microsoft Windows App Studio, OU=Microsoft Windows App Studio
[02:15:02.003 - Verbose]::Templates content extracted to C:\Users\crutkas\AppData\Local\Temp\w1544taf.cy1.
[02:15:06.907 - Verbose]::Next automatic content verification at 7/5/2017 4:15:02 PM
[02:15:06.908 - Verbose]::Next automatic content verification at 7/5/2017 4:15:02 PM
[02:15:10.081 - Verbose]::Next automatic content verification at 7/5/2017 4:15:02 PM
[02:15:10.081 - Verbose]::Next automatic content verification at 7/5/2017 4:15:02 PM
[02:15:21.798 - Verbose]::Next automatic content verification at 7/5/2017 4:15:02 PM
[02:15:21.798 - Verbose]::Next automatic content verification at 7/5/2017 4:15:02 PM
[02:17:44.289 - Verbose]::Templates content downloaded to C:\Users\crutkas\AppData\Local\Temp\vyqtyoqt.3t1\Templates.mstx from https://wtsrepository.blob.core.windows.net/dev/Latest/Templates.mstx.
[02:17:44.315 - Verbose]::Package certificate OID.0.9.2342.19200300.100.1.1=10018646, CN=Microsoft Windows App Studio, OU=Microsoft Windows App Studio
[02:17:44.319 - Verbose]::Templates content downloaded to C:\Users\crutkas\AppData\Local\Temp\hcpz2nnp.dgn\Templates.mstx from https://wtsrepository.blob.core.windows.net/dev/Latest/Templates.mstx.
[02:17:44.352 - Verbose]::Package certificate OID.0.9.2342.19200300.100.1.1=10018646, CN=Microsoft Windows App Studio, OU=Microsoft Windows App Studio
[02:17:45.270 - Verbose]::Templates content extracted to C:\Users\crutkas\AppData\Local\Temp\ayxe1kqx.wvp.
[02:17:45.275 - Verbose]::Templates content extracted to C:\Users\crutkas\AppData\Local\Temp\5lyfcqvi.i3a.
[02:17:45.493 - Warning]::The folder C:\Users\crutkas\AppData\Local\Temp\5lyfcqvi.i3a can't be delete. Error: Access to the path '_postaction.csproj' is denied.
----------- Addtional Exception Info -----------
System.UnauthorizedAccessException: Access to the path '_postaction.csproj' is denied.
   at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
   at Microsoft.Templates.Core.Fs.SafeDeleteDirectory(String dir)
-----------------------------------------------
[02:17:45.501 - Warning]::The folder C:\Users\crutkas\AppData\Local\Temp\5lyfcqvi.i3a can't be delete. Error: Access to the path 'description.md' is denied.
----------- Addtional Exception Info -----------
System.UnauthorizedAccessException: Access to the path 'description.md' is denied.
   at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
   at Microsoft.Templates.Core.Fs.SafeDeleteDirectory(String dir)
-----------------------------------------------
[02:17:45.722 - Warning]::The folder C:\Users\crutkas\AppData\Local\Temp\ayxe1kqx.wvp can't be delete. Error: Access to the path 'description.md' is denied.
----------- Addtional Exception Info -----------
System.UnauthorizedAccessException: Access to the path 'description.md' is denied.
   at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
   at Microsoft.Templates.Core.Fs.SafeDeleteDirectory(String dir)
-----------------------------------------------
[02:17:45.726 - Warning]::The folder C:\Users\crutkas\AppData\Local\Temp\ayxe1kqx.wvp can't be delete. Error: Access to the path 'ActivationService_postaction.cs' is denied.
----------- Addtional Exception Info -----------
System.UnauthorizedAccessException: Access to the path 'ActivationService_postaction.cs' is denied.
   at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
   at Microsoft.Templates.Core.Fs.SafeDeleteDirectory(String dir)
-----------------------------------------------
[02:17:45.732 - Error]::Microsoft.Templates.Core.RepositorySynchronizationException: Error syncing templates. Status: 'Acquiring' ---> System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   at System.Collections.Generic.List`1.Enumerator.MoveNext()
   at Microsoft.Templates.Core.Locations.TemplatesSource.CleanUpTemps()
   at Microsoft.Templates.Core.Locations.TemplatesSource.Extract(String mstxFilePath, String targetFolder)
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.AcquireContent()
   --- End of inner exception stack trace ---
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.AcquireContent()
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.<AcquireContentAsync>b__28_0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.<AcquireContentAsync>d__28.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.<CheckForNewContentAsync>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.Core.TemplatesRepository.<CheckForUpdatesAsync>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.UI.ViewModels.Common.BaseMainViewModel.<OnCheckUpdates>d__94.MoveNext()
========== Tracked Exception [20170705 02:17:45.755] ==========
Microsoft.Templates.Core.RepositorySynchronizationException: Error syncing templates. Status: 'Acquiring' ---> System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   at System.Collections.Generic.List`1.Enumerator.MoveNext()
   at Microsoft.Templates.Core.Locations.TemplatesSource.CleanUpTemps()
   at Microsoft.Templates.Core.Locations.TemplatesSource.Extract(String mstxFilePath, String targetFolder)
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.AcquireContent()
   --- End of inner exception stack trace ---
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.AcquireContent()
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.<AcquireContentAsync>b__28_0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.<AcquireContentAsync>d__28.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.Core.Locations.TemplatesSynchronization.<CheckForNewContentAsync>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.Core.TemplatesRepository.<CheckForUpdatesAsync>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Templates.UI.ViewModels.Common.BaseMainViewModel.<OnCheckUpdates>d__94.MoveNext()
--------------------------------------------------------------
[02:19:04.057 - Verbose]::Templates content downloaded to C:\Users\crutkas\AppData\Local\Temp\dkddkeup.ai1\Templates.mstx from https://wtsrepository.blob.core.windows.net/dev/Latest/Templates.mstx.
[02:19:04.084 - Verbose]::Package certificate OID.0.9.2342.19200300.100.1.1=10018646, CN=Microsoft Windows App Studio, OU=Microsoft Windows App Studio
[02:19:04.940 - Verbose]::Templates content extracted to C:\Users\crutkas\AppData\Local\Temp\hylryp4h.ptp.
```

**Comment by ralarcon** [comment]
Fixed in PR #638

**Comment by dgomezc** [comment]
Checked on version:
Wizard: 0.14.17244.65002
Templates: 0.14.17244.2
